internal/services/network/subnet_data_source.go

```diff
@@ -135,6 +141,10 @@ func dataSourceSubnetRead(d *pluginsdk.ResourceData, meta interface{}) error {
 				d.Set("enforce_private_link_service_network_policies", flattenEnforceSubnetNetworkPolicy(string(*props.PrivateLinkServiceNetworkPolicies)))
 			}
 
+			if props.DefaultOutboundAccess != nil {
+				d.Set("default_outbound_access_enabled", props.DefaultOutboundAccess)
+			}
```

Comment:
we need to ensure a value is always set into the state, since this currently defaults to `true` in the API (for previous resources) can we default this on our side when it's not returned:

```suggestion
			defaultOutboundAccessEnabled := true
			if props.DefaultOutboundAccess != nil {
				defaultOutboundAccessEnabled = *props.DefaultOutboundAccess
			}
			d.Set("default_outbound_access_enabled", defaultOutboundAccessEnabled)
```